### DISCUSSION TITLE
Redesign look and feel of website to be much simpler

### DISCUSSION THREAD

**Discussion by wd15** [usnistgov/pfhub#1413] [General]
Simple design based on one of the simple Jekyll themes - https://jekyllthemes.io/free

Hyde is my preference: https://jekyllthemes.io/theme/hyde

**Comment by tkphd** [comment]
Broadly in favor of adopting a Jekyll theme. Let's discuss how we want the website to be used before down-selecting a price point or specific theme.

## [JekyllThemes.io](https://jekyllthemes.io) 

A bit tedious to navigate, but free & paid offerings are divided into broad categories:

### [Portfolio](https://jekyllthemes.io/jekyll-portfolio-themes)

> Show off your work with these awesome image-focused portfolio themes. Perfect for designers, developers, photographers and illustrators.
  
### [Blog](https://jekyllthemes.io/jekyll-blog-themes)

> Start a personal or business blog with these handpicked templates that are great for displaying your posts in a beautiful, organized way.

### [Business](https://jekyllthemes.io/jekyll-business-themes)

> Need a great landing page for your business, product or service? We've found the best templates to get you started.

## [JamStack Themes](https://jamstackthemes.dev/ssg/jekyll/)

Just a bunch of themes with no apparent organization or filtering, but each tile links to the repo & demo site.

## [JekyllThemes.org](http://jekyllthemes.org)

Likewise, just a bunch of themes, with substantial overlap with JamStack and a *search bar!*

## [CloudCannon Themes](https://cloudcannon.com/community/themes/)

e-commerce-centric themes, which might be what we're looking for?

**Comment by wd15** [comment]
Regarding your question of how we want the website to be used. Primarily we want to show off Jupyter Notebooks as some of the other content will likely be moved to the usnistgov/pfhub wiki. We want minimal CSS with notebooks as it tends to impact the layout of the notebook. We want to show some meta data about PFHub (this most likely will be on a notebook as well). We may also want to have a guide for phase field at some point that is more navigable than the wiki. I'm thinking that a documentation style might be the way to go. These are minimal and easily navigable and might not interfere with displaying notebooks. Some possibilities,

 - [Just the docs to keep a have a software documentation style which might be good with notebooks](https://jekyllthemes.io/theme/just-the-docs)
 - [very minimal theme](https://jekyllthemes.io/theme/alembic)
 - [hyde is pretty minimal](https://jekyllthemes.io/theme/hyde)
 - [another documentation theme](https://jekyllthemes.io/theme/documentation)

**Comment by tkphd** [comment]
What do you think about [Minimal Mistakes](https://github.com/mmistakes/minimal-mistakes)? Of note, it offers layouts for a few page types that would be useful for us (single, archive index, search, splash, and paginated home page).

**Reply by wd15** [reply]
Looks fine. It's minimal which is exactly what we want. I think a splash is good just as we already have one although I really don't care for it anymore. Thing to do is to do a conversion over to that and then let's ask others. I'll try and make some progress on that once we have a settled upload scheme, which is slightly higher priority currently.

**Comment by wd15** [comment]
I'm wondering if Jupyter Book might be a good alternative -- https://jupyterbook.org/en/stable/intro.html -- since we're hosting a lot of content with Notebooks. It's an alternative to Jekyll, but handles the notebooks better than our current configuration.

**Reply by tkphd** [reply]
Possibly? I think JupyterBook is a good idea for "publishing" our Best Practices after they've been gathered in the Wiki. I'm not sure the constraints of a book format are compatible with what we want the website as a whole to do, i.e., present all the various facets of the PFHub effort in a clear and meaningful way. Books make sense for a single topic, but PFHub spans several: Benchmark definitions, solutions, phase-field theory & education, invited essays & news, etc.

**Reply by wd15** [reply]
I'm not sure Jupyter Book needs to be a book. It's just a nice way of turning notebooks (along with *.md and *.rst) into static content. I'd have to play with it a bit too see.

**Reply by wd15** [reply]
Currently, we have a makefile that turns the notebooks into markdown and then lots of hacks to deal with conflicting CSS and other annoying things. It basically just handles notebooks on the web much better. I might try and throw something together as a prototype and then we can have a much more informed discussion.

**Reply by wd15** [reply]
One annoying thing about the current approch is clashes between Liquid templating and using templating in the notebooks. I think this would probably be better with Jupyter Book. Jekyll can handle a lot of data processing. I'm not sure if Jupyter Book can do that though. That might be a downside. It might be possible to use both together actually.
